src/antworld/surface.cc

```diff
@@ -0,0 +1,90 @@
+#include "antworld/surface.h"
+#include "antworld/texture.h"
+
+//----------------------------------------------------------------------------
+// BoBRobotics::AntWorlds::Surfaces
+//----------------------------------------------------------------------------
+namespace BoBRobotics
+{
+namespace AntWorld
+{
+Surface::Surface() 
+:   m_PositionVBO(0), m_ColourVBO(0), m_TexCoordVBO(0), m_IBO(0), 
+    m_NumVertices(0), m_NumIndices(0), m_Texture(nullptr)
+{
+    // Create a vertex array object to bind everything together
+    glGenVertexArrays(1, &m_VAO);
+}
+//----------------------------------------------------------------------------
+Surface::~Surface()
+{
+    if(m_PositionVBO != 0) {
+        glDeleteBuffers(1, &m_PositionVBO);
+    }
+
+    if(m_ColourVBO != 0) {
+        glDeleteBuffers(1, &m_ColourVBO);
+    }
+
+    if(m_TexCoordVBO != 0) {
+        glDeleteBuffers(1, &m_TexCoordVBO);
+    }
+    
+    if(m_IBO != 0) {
+        glDeleteBuffers(1, &m_IBO);
+    }
+
+    glDeleteVertexArrays(1, &m_VAO);
+}
+//----------------------------------------------------------------------------
+void Surface::bind(bool bindTexture) const
+{
+    // Bind world VAO
+    glBindVertexArray(m_VAO);
+
+    // If surface has a texture, bind it
+    if(bindTexture) {
+        if(m_Texture != nullptr) {
+            glEnable(GL_TEXTURE_2D);
+            m_Texture->bind();
+        }
+        // Otherwise make sure no textures are bound
+        else {
+            glBindTexture(GL_TEXTURE_2D, 0);
+        }
+    }
+}
+//----------------------------------------------------------------------------
+void Surface::unbind(bool unbindTexture) const
+{
+    // If surface has a texture, bind it
+    if(unbindTexture && m_Texture != nullptr) {
+        glDisable(GL_TEXTURE_2D);
+        m_Texture->unbind();
+    }
+
+    // Unbind vertex array
+    glBindVertexArray(0);
+}
```

Comment:
It might be clearer in calling code if these were split into separate functions instead, e.g. ``bind()`` and ``bindWithTexture()``, but up to you.

Reply:
agreed!

Reply:
:+1: